main.queues.tf

```diff
@@ -1,30 +1,32 @@
-resource "azurerm_storage_queue" "this" {
+resource "azapi_resource" "queue" {
   for_each = var.queues
 
-  name                 = each.value.name
-  storage_account_name = azurerm_storage_account.this.name
-  metadata             = each.value.metadata
+  type = "Microsoft.Storage/storageAccounts/queueServices/queues@2023-01-01"
+  body = jsonencode({
+    properties = {
+      # metadata = each.value.metadata
+    }
+  })
+  name                      = each.value.name
+  parent_id                 = "${azurerm_storage_account.this.id}/queueServices/default"
+  schema_validation_enabled = false
 
   dynamic "timeouts" {
     for_each = each.value.timeouts == null ? [] : [each.value.timeouts]
     content {
       create = timeouts.value.create
       delete = timeouts.value.delete
       read   = timeouts.value.read
-      update = timeouts.value.update
```

Comment:
The `update` timeout is removed in `azapi v1.13.1` by mistake, we'll support it in the next release.